src/Microsoft.Framework.Configuration/Constants.cs

```diff
@@ -5,6 +5,6 @@ namespace Microsoft.Framework.Configuration
 {
     public static class Constants
     {
-        public static readonly string KeyDelimiter = ":";
+        public static readonly string[] KeyDelimiter = new[] { ":" };
```

Comment:
Thanks @divega for your feedback, I was waiting for such feedback from two months ago :smile: 
I will update the PR soon